libpod/container_internal_linux.go

```diff
@@ -279,6 +279,17 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
+	hasHomeSet := false
+	for _, s := range c.config.Spec.Process.Env {
+		if strings.HasPrefix(s, "HOME=") {
```

Comment:
Would having home set to `""` break this? Should we try and detect that?

Reply:
not sure, I think if it is set (as in the `HOME=` case) we should honor that and not try to override it, this is what I get now with the PR:
```
$ bin/podman run --rm -it fedora sh -c 'echo HOME=$HOME'
HOME=/root
$ bin/podman run --rm -it --env HOME= fedora sh -c 'echo HOME=$HOME'
HOME=
```

Reply:
SGTM